Clarifying roles required to setup Azure DevOps (#5161)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/git/setup-azure.md

---
title: "Set up Azure DevOps"
id: "setup-azure"
description: "You can set up your Azure DevOps by creating an Entra ID app and adding it to dbt Cloud."
sidebar_label: "Set up Azure DevOps"
---

<Snippet path="available-enterprise-tier-only" />

## Overview

To use our native integration with Azure DevOps in dbt Cloud, an account admin needs to set up an Microsoft Entra ID app. We recommend setting up a separate [Entra ID application than used for SSO](/docs/cloud/manage-access/set-up-sso-azure-active-directory).

1. [Register an Entra ID app](#register-an-entra-id-app).
2. [Add permissions to your new app](#add-permissions-to-your-new-app).
3. [Add another redirect URI](#add-another-redirect-uri).
4. [Connect Azure DevOps to your new app](#connect-azure-devops-to-your-new-app).
5. [Add your Entra ID app to dbt Cloud](#add-your-azure-ad-app-to-dbt-cloud).

> Once the Entra ID app is added to dbt Cloud, an account admin must also [connect a service user]((#connecting-a-service-user)) via OAuth, which will be used to power headless actions in dbt Cloud such as deployment runs and CI.

Author: ```suggestion
Once the Entra ID app is added to dbt Cloud, an account admin must also [connect a service user](#connecting-a-service-user) via OAuth, which will be used to power headless actions in dbt Cloud such as deployment runs and CI.
```

Once the Entra ID app is added to dbt Cloud and the service user is connected, then dbt Cloud developers can personally authenticate in dbt Cloud from Azure DevOps. For more on this, see [Authenticate with Azure DevOps](/docs/cloud/git/authenticate-azure).

The following personas are required to complete the steps on this page:
- Entra ID admin
- Azure DevOps admin
- dbt Cloud account admin

## Register an Entra ID app

> An Entra ID account admin needs to perform the following steps:

Author: ```suggestion
An Entra ID admin needs to perform the following steps:
```

An Entra ID admin needs to add another redirect URI to your Entra ID application. This redirect URI will be used to authenticate the service user for headless actions in deployment environments.

1. Navigate to your Entra ID application.
2. Select the link next to **Redirect URIs**
3. Click **Add URI** and add the URI, replacing `YOUR_ACCESS_URL` with the [appropriate Access URL](/docs/cloud/about-cloud/access-regions-ip-addresses) for your region and plan:
`https://YOUR_ACCESS_URL/complete/azure_active_directory_service_user`
4. Click **Save**.

<Lightbox src="/img/docs/dbt-cloud/connecting-azure-devops/redirect-uri.gif" title="Adding the Service User redirect URI"/>

## Create a client secret

An Entra ID admin needs to complete the following steps:

1. Navigate to your Entra ID application.
2. Select **Certificates and Secrets** from the left navigation panel.
3. Select **Client secrets** and click **New client secret**
4. Give the secret a description and select the expiration time. Click **Add**.
5. Copy the **Value** field and securely share it with the dbt Cloud account admin who will complete the setup. 

## Connect Azure DevOps to your new app

> An Azure admin will need one of the following permissions in both the Entra ID and Azure DevOps environments:

Reviewer: Is "Azure admin" another persona? Or is ti the same as one of the others?

If it is not, can we add it at the top in the list of personas a(and maybe mention what sort of admin and/or access it requires)

Author: @b-per Sure, I'll add it up top.

You must connect your service user before setting up a dbt Cloud project, as the service user's permissions determine which projects dbt Cloud can import.

> A dbt Cloud account admin with access to the service users Azure DevOps account must complete the following to connect the service user:

Author: ```suggestion
A dbt Cloud account admin with access to the service user's Azure DevOps account must complete the following to connect the service user:
```

> 1. Sign into the service user's Azure DevOps account.

Author: ```suggestion
1. Sign in to the service user's Azure DevOps account.
```

To connect the service user:

> 1. An admin must first be signed into the service user's Azure DevOps account.

Reviewer: Which Admin is it in that case?
